napari/components/_viewer/view/main.py

```diff
@@ -28,20 +28,20 @@ def __init__(self, viewer):
         self.canvas.connect(self.on_mouse_release)
         self.canvas.connect(self.on_key_press)
         self.canvas.connect(self.on_key_release)
+        self.canvas.connect(self.on_draw)
 
         self.view = self.canvas.central_widget.add_view()
 
-        # TO DO: Remove
-        self.viewer._scene = self.view.scene
-
         # Set 2D camera (the camera will scale to the contents in the scene)
         self.view.camera = PanZoomCamera(aspect=1)
         # flip y-axis to have correct aligment
         self.view.camera.flip = (0, 1, 0)
         self.view.camera.set_range()
-
         self.view.camera.viewbox_key_event = viewbox_key_event
 
+        # TO DO: Remove
```

Comment:
Hmmmmm...

Reply:
yeah, this is needed until we pull the vispy stuff out of each layer's model file and into it's own view file. i'll work on that after you remove `add_to_viewer`